src/main/java/io/github/nihaljain/Util.java

```diff
@@ -40,16 +41,12 @@ public static String getLogDirPath(String[] args) throws Exception {
     logger.info("useCanonicalPath flag: " + useCanonicalPath);
 
     // Get the final log path for the log directory
-    String logDirPath = useCanonicalPath ? Paths.get(logDir).toFile().getCanonicalPath() : logDir;
+    Path logDirPath = useCanonicalPath ? Paths.get(logDir).toRealPath() : Paths.get(logDir);
     logger.info("Logs directory (final path): " + logDirPath);
 
     // Ensure the log directory exists, else create it
-    File logDirectory = new File(logDir);
-    if (!logDirectory.exists() || !logDirectory.isDirectory()) {
-      if (!logDirectory.mkdirs()) {
-        throw new IllegalArgumentException(
-            "Log directory does not exist and could not be created: " + logDir);
-      }
+    if (!Files.exists(logDirPath) || !Files.isDirectory(logDirPath)) {
+      Files.createDirectories(logDirPath);
```

Comment:
```suggestion
    if (!Files.isDirectory(logDirPath)) {
      Files.createDirectories(logDirPath);
```

The `isDirectory` check ensures that it exists too.
If it didn't exist, the `isDirectory` check returns false.